### ISSUE TITLE
WebGPU

### ISSUE THREAD

**Issue by mrhegemon** [XRFoundation/XREngine#5489]
![Messages Image(58322290)](https://user-images.githubusercontent.com/5104160/158294981-21ae5557-ea59-408d-affc-f58fedd2c381.png)

Turn it on in Three VS ECS Renderer.....

https://web.dev/gpu-compute/

https://github.com/gpuweb/gpuweb

https://threejs.org/examples/?q=webgpu

WebGPU is available for now in Chrome Canary on desktop behind an experimental flag. You can enable it at chrome://flags/#enable-unsafe-webgpu. The API is constantly changing and currently unsafe. As GPU sandboxing isn't implemented yet for the WebGPU API, it is possible to read GPU data for other processes! Don't browse the web with it enabled.

https://github.com/gpuweb/types

https://www.i-programmer.info/news/144-graphics-and-games/15236-safari-completes-support-for-webgl-2.html

https://github.com/maierfelix/webgpu
https://dawn.googlesource.com/dawn/
https://webkit.org/status/#?search=webgpu

https://austin-eng.com/webgpu-samples/

Deno is so hot... Just Sayin...
https://www.infoq.com/news/2021/03/deno-1-8-webgpu/

**Comment by tomzhang** [comment]
i think this funcationality is very important ,because think about a large scense with a group of people display with interoperable on the client side . the smart phone's  system will crash out and use expirence will be a disaster without muti-thread drawing

**Comment by HexaField** [comment]
> i think this funcationality is very important ,because think about a large scense with a group of people display with interoperable on the client side . the smart phone's system will crash out and use expirence will be a disaster without muti-thread drawing

I agree! We have this on our roadmap, we are waiting for the technology and ecosystem to mature a little bit more, and we have other features that will enable this to be used efficiently with our ECS.

**Comment by tomzhang** [comment]
@HexaField  i came from china with company named "妙奇互娱",  i would like be your project's Business supporter with using XREngine as a port of solution . i would like pay as unity's way when my company's income increasing to 1M USD per year i would like pay your team 4K USD per month .

**Comment by tomzhang** [comment]
also later i will hiring some people who familiar with open source and rendering engine that will dedicated to working with this project with yours
